lib/src/date-input/DatePicker.tsx

```diff
@@ -0,0 +1,205 @@
+  const ref = useRef(null);
+  const translatedLabels = useTranslatedLabels();
+
+  const handleKeyboardEvent = (event) => {
+    if (event.key === "Escape") {
+      event.preventDefault();
+      onEscCalendar();
+    }
+  };
+  const handleClickOutside = (event) => {
+    if (ref?.current && !ref?.current.contains(event.target)) {
+      onCloseCalendar();
+    }
+  };
+
+  useEffect(() => {
+    document.addEventListener("mousedown", handleClickOutside);
+    document.addEventListener("keydown", handleKeyboardEvent);
+    return () => {
+      document.removeEventListener("mousedown", handleClickOutside);
+      document.removeEventListener("keydown", handleKeyboardEvent);
+    };
+  }, [onCloseCalendar]);
+
+  const handleDateSelect = (date: Dayjs) => {
+    setInnerDate(date);
+    onDateSelect(date);
+  };
+
+  const handleMonthChange = (date: Dayjs) => {
+    setInnerDate(date);
+  };
+
+  return (
+    <DatePicker id={id} ref={ref}>
+      <PickerHeader>
+        <HeaderButton
+          aria-label={translatedLabels.calendar.previousMonthTitle}
+          title={translatedLabels.calendar.previousMonthTitle}
+          onClick={() => handleMonthChange(innerDate.set("month", innerDate.get("month") - 1))}
+        >
+          {leftCaret}
+        </HeaderButton>
+        <HeaderYearTrigger
+          aria-live="polite"
+          onClick={() => setContent((content) => (content === "yearPicker" ? "calendar" : "yearPicker"))}
+        >
+          <HeaderYearTriggerLabel>
+            {translatedLabels.calendar.months[innerDate.get("month")]} {innerDate.format("YYYY")}
+          </HeaderYearTriggerLabel>
+          {downCaret}
+        </HeaderYearTrigger>
+        <HeaderButton
+          aria-label={translatedLabels.calendar.nextMonthTitle}
+          title={translatedLabels.calendar.nextMonthTitle}
+          onClick={() => handleMonthChange(innerDate.set("month", innerDate.get("month") + 1))}
+        >
+          {rightCaret}
+        </HeaderButton>
+      </PickerHeader>
+      {content === "calendar" && (
+        <Calendar
+          innerDate={innerDate}
+          selectedDate={selectedDate}
+          onInnerDateChange={setInnerDate}
+          onDaySelect={handleDateSelect}
+        />
+      )}
+
+      {content === "yearPicker" && (
+        <YearPicker
+          onYearSelect={(year) => {
+            setInnerDate(innerDate.set("year", year));
+            setContent("calendar");
+          }}
+          selectedDate={selectedDate}
+        />
+      )}
+    </DatePicker>
+  );
+};
+
+const DatePicker = styled.div`
+  padding-top: 16px;
+  background-color: ${(props) => props.theme.dateInput.pickerBackgroundColor};
+  box-shadow: 0 4px 6px -1px rgba(0, 0, 0, 0.1);
+  border: 1px solid #bfbfbf;
+  border-radius: 4px;
+`;
+
+const PickerHeader = styled.div`
+  display: flex;
+  gap: 8px;
+  align-items: center;
+  justify-content: space-between;
+  box-sizing: border-box;
+  height: 40px;
```

Comment:
This height is unnecessary since it is already being defined by the height of the `HeaderYearTrigger`

Reply:
Removed.